addon/views/header-cell.js

```diff
@@ -0,0 +1,148 @@
+import Ember from 'ember';
+import StyleBindingsMixin from 'ember-table/mixins/style-bindings';
+import RegisterTableComponentMixin from 'ember-table/mixins/register-table-component';
+
+export default Ember.View.extend(
+StyleBindingsMixin, RegisterTableComponentMixin, {
+  // ---------------------------------------------------------------------------
+  // API - Inputs
+  // ---------------------------------------------------------------------------
+
+  // TODO: Doc
+  templateName: 'header-cell',
+  classNames: ['ember-table-cell', 'ember-table-header-cell'],
+  classNameBindings: ['column.isSortable:sortable', 'column.textAlign'],
+  styleBindings: ['width', 'height'],
+
+  // ---------------------------------------------------------------------------
+  // Internal properties
+  // ---------------------------------------------------------------------------
+
+  column: Ember.computed.alias('content'),
+  width: Ember.computed.alias('column.width'),
+  minWidth: Ember.computed.alias('column.minWidth'),
+  maxWidth: Ember.computed.alias('column.maxWidth'),
+  nextResizableColumn: Ember.computed.alias('column.nextResizableColumn'),
+  height: Ember.computed.alias('tableComponent._headerHeight'),
+
+  effectiveMinWidth: Ember.computed(function() {
+    if (this.get('tableComponent.columnMode') === 'standard') {
+      return this.get('minWidth');
+    }
+    var nextColumnMaxDiff = this.get('nextResizableColumn.maxWidth') -
+        this.get('nextResizableColumn.width');
+    if (this.get('minWidth') && nextColumnMaxDiff) {
+      return Math.min(this.get('minWidth'), this.get('width') -
```

Comment:
I usually prefer to store the return value in one variable and have a single `return` statement when possible. It's easier to read IMO.